src/TestSuite/Fixture/FixtureLoader.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\TestSuite\Fixture;
+
+use Cake\Datasource\ConnectionInterface;
+use Cake\TestSuite\TestCase;
+
+/**
+ * Abstract base class and singleton container for fixture
+ * managers.
+ */
+abstract class FixtureLoader
+{
+    /**
+     * @var \Cake\TestSuite\Fixture\FixtureLoader|null
+     */
+    private static $instance;
+
+    /**
+     * Set the shared instance
+     *
+     * @param \Cake\TestSuite\Fixture\FixtureLoader $instance The instance to set.
+     * @return void
+     */
+    public static function setInstance(FixtureLoader $instance): void
```

Comment:
Can we create a factory method and a protected constructor or does it have to be 'set`?

Reply:
If the constructor was protected how would each fixture system set the manager it needs? I don't love the approach I have here but it was simple and would let me move the overall implementation forward.